### PR TITLE
Feature/details page

### PR DESCRIPTION
**Pull request by marzad**

### CODE REVIEW

**Review by MoskaFan — APPROVED** [review]
Nice job

File: frontend/src/components/AddBook.tsx [hunk 1]
```diff
@@ -15,27 +15,29 @@ export default function AddBook(props: AddBookProps){
...
         <section>
             <h1>New Book:</h1>
-            Title: <input onChange={newBookTitleInput}/>
-            Author: <input onChange={newBookAuthorInput}/>
-            ISBN: <input onChange={newBookIsbnInput}/>
+            Title: <input onChange={newBookTitleInput} value={newBookTitle}/>
+            Author: <input onChange={newBookAuthorInput} value={newBookAuthor}/>
+            ISBN: <input onChange={newBookIsbnInput} value={newBookIsbn}/>
             <button onClick={addBook}>Add</button>
         </section>
```
Comment (Flooooooooooorian) [inline frontend/src/components/AddBook.tsx:42]: I would use a <from> here

File: frontend/src/components/SearchForAuthor.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        props.inputFieldValue(inputValue)
+        setInputValue("")
+    }
+
+    return(
+        <>
+            Author name:
+            <input onChange={getInputFieldValue} value={inputValue}/>
+            <button onClick={setInputFieldValue}>search</button>
+        </>
```
Comment (Flooooooooooorian) [inline frontend/src/components/SearchForAuthor.tsx:25]: I would use a form here aswell

File: frontend/src/hooks/UseBooks.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    bookList: BookData[],
+    addBook: (book: BookData) => void,
+    deleteBook: (id: string) => void,
+    getBookByID: (id: string) => void,
+    getBookByISBN: (isbn: string) => void,
+    getBookByAuthor: (name: string) => void,
+    getBookByKeyword: (keyword: string) => void
+}
+
+export default function UseBooks(): UseBooksReturn{
```
Comment (Flooooooooooorian) [inline frontend/src/hooks/UseBooks.tsx:15]:
```suggestion
export default function useBooks(): UseBooksReturn{
```

File: frontend/src/hooks/UseBooks.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            .catch(console.error)
+    }
+
+    function deleteBook(deletedId: string){
+        axios.delete("/books/" + deletedId)
+            .then(()=>{
+                const newList = bookList.filter((book: BookData)=>
+                    book.id!==deletedId)
+                setBookList(newList)
+            })
```
Comment (Flooooooooooorian) [inline frontend/src/hooks/UseBooks.tsx:41]:
```suggestion
            })
            .catch(console.error)
```

File: frontend/src/hooks/UseBooks.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    function getBookByID(id: string){
+        axios.get("/books/" + id)
+            .then(() => {
+                const filteredBook = bookList.filter(book => {
+                        return book.id === id
+                    })
+                if(id){
+                    setBookList(filteredBook)
+                }
+            })
```
Comment (Flooooooooooorian) [inline frontend/src/hooks/UseBooks.tsx:53]:
```suggestion
            })
            .catch(console.error)
```

File: frontend/src/hooks/UseBooks.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    function getBookByKeyword(keyword: string){
+        axios.get("/books/by-keyword/?keyword=" + keyword)
+            .then(() => {
+                const filteredBookByTitle = bookList.filter(book => {
+                        return book.title.toLowerCase().includes(keyword.toLowerCase())
+                    })
+                setBookList(filteredBookByTitle)
+            })
```
Comment (Flooooooooooorian) [inline frontend/src/hooks/UseBooks.tsx:64]:
```suggestion
            })
            .catch(console.error)
```

File: frontend/src/hooks/UseBooks.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    function getBookByISBN(isbn: string){
+        axios.get("/books/by-isbn/?isbn=" + isbn)
+            .then(() => {
+                const filteredBookByISBN = bookList.filter(book => {
+                        return book.isbn.toLowerCase().includes(isbn.toLowerCase())
+                    })
+                setBookList(filteredBookByISBN)
+            })
```
Comment (Flooooooooooorian) [inline frontend/src/hooks/UseBooks.tsx:75]:
```suggestion
            })
            .catch(console.error)
```

File: frontend/src/hooks/UseBooks.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+    function getBookByAuthor(name: string){
+        axios.get("/books/by-author/?name=" + name)
+            .then(() => {
+                const filteredBookByAuthor =
+                    bookList.filter(book => {
+                        return book.author.toLowerCase().includes(name.toLowerCase())
+                    })
+                setBookList(filteredBookByAuthor)
+            })
```
Comment (Flooooooooooorian) [inline frontend/src/hooks/UseBooks.tsx:87]:
```suggestion
            })
            .catch(console.error)
```

File: frontend/src/hooks/UseBooks.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    function getBookByAuthor(name: string){
+        axios.get("/books/by-author/?name=" + name)
+            .then(() => {
+                const filteredBookByAuthor =
+                    bookList.filter(book => {
+                        return book.author.toLowerCase().includes(name.toLowerCase())
+                    })
+                setBookList(filteredBookByAuthor)
```
Comment (Flooooooooooorian) [inline frontend/src/hooks/UseBooks.tsx:86]: If you have an Endpoint which returns only books with the given author why do you need to filter the list again?

File: books.json [hunk 10]
```diff
@@ -0,0 +1,92 @@
+{
```
Comment (Flooooooooooorian) [inline books.json:1]: Why do you need any placeholder data? The data should be saved in the backend only

**Review by Flooooooooooorian — APPROVED** [review]

**Pull request merged**